File: LMS/LMS.UI/huda-center/src/app/shared/models/library-management/products/product-update-request.dto.ts
// src/app/shared/models/library-management/products/product-update-request.dto.ts

export interface ProductUpdateRequestDto {
  arProductName: string;
  arProductDescription: string;
  enProductName: string;
  enProductDescription: string;
  productPrice: number;
  productStock: number;
  categoriesIds: string[];
  imageFile: File | null; // لرفع الملف (يمكن أن يكون null إذا لم يتم تغيير الصورة)
}

File: LMS/LMS.API/DTOs/LibraryManagement/Products/ProductUpdateRequestDto.cs
using Microsoft.AspNetCore.Mvc;

namespace LMS.API.DTOs.LibraryManagement.Products
{
    public class ProductUpdateRequestDto
    {
        public string ARProductName { get; set; }

        public string ARProductDescription { get; set; }

        public string ENProductName { get; set; }

        public string ENProductDescription { get; set; }

        public decimal ProductPrice { get; set; }

        public int ProductStock { get; set; }


        public ICollection<Guid> CategoriesIds { get; set; } = [];

        [FromForm]
        public IFormFile? ImageFile { get; set; }
    }
}
